clusterbuster

```diff
@@ -2351,6 +2354,91 @@ EOF
     done
 }
 
+function create_vm_deployment() {
+    local namespace=$1
+    local instance=$2
+    local -i replicas=$4
+    local -i replica=0
+    local depname="${namespace}-${workload}-${instance}"
+    while (( replica++ < replicas )) ; do
+        local name="${depname}-${replica}"
+        local vmname=$(mkpodname "$name")
+        create_object -n "$namespace" "$name" <<EOF
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  labels:
+    app: $name
+    kubevirt-vm: $vmname
+  name: $vmname
+$(indent 2 standard_deployment_metadata_yaml "$namespace" client)
+$(indent 2 standard_pod_metadata_yaml "$namespace" client)
+$(create_vm_spec -n "$vmname")
+EOF
+    done
+}
+
+function create_vm_spec() {
+    local image
+    local vmname
+    local cores
+    local sockets
+    local threads
+    local mem
+    local grace
+    local OPT
+    local OPTIND=1
+    local OPTARG
+    while getopts ':n:c:s:t:m:g:i:-' OPT; do
+        case "$OPT" in
+        '?') fatal "Unknown option -$OPTARG" ;;
+         : ) fatal "Option -$OPTARG requires a value" ;;
+         - ) break ;; # end options with --
```

Comment:
This isn't necessary here.  getopts will always terminate parsing when it sees a `--` argument.  The only reason to accept `-` is to do long name parsing, e. g. `--variable=value`.

Also, please remove the space between the option character and the parenthesis (just for common style).